src/Message/Body.php

```diff
@@ -22,13 +30,23 @@
     public function getBody()
     {
         if (!isset($this->body)) {
-            $this->body = Stream::open($this->defaultStream, $this->defaultMode);
+            $this->body = $this->createDefaultBody();
```

Comment:
A trait shouldn't use properties it isn't sure will exist. It's better to use a method instead of properties, since we can simply add an abstract method to the trait.